server/apiserver/apiserver.go

```diff
@@ -74,3 +80,74 @@ func (api *ApiServer) Headers(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 }
+
+// Upgrader "upgrades" HTTP endpoint to WS endpoint, this requires a Read and Write buffer size
+var upgrader = websocket.Upgrader{} // use default options
+
+// define a reader which will listen for new messages being sent to our WebSocket endpoint
+func reader(conn *websocket.Conn) {
+	for {
+		// read in a message
+		// messageType is an int with value websocket.BinaryMessage or websocket.TextMessage
+		// p is []byte
+		messageType, p, err := conn.ReadMessage()
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		fmt.Println(messageType) // to get rid of unused variable
+
+		// holds a map of strings to arbitrary data types
+		var dat map[string]interface{}
+
+		if err := json.Unmarshal(p, &dat); err != nil {
+			log.Printf("error decoding client response: %v", err)
+			if e, ok := err.(*json.SyntaxError); ok {
+				log.Printf("syntax error at byte offset %d", e.Offset)
+			}
+			log.Printf("client response: %q", p)
+			panic(err)
+		}
+		fmt.Println(dat)
+
+		// convert each attribute to appropriate type
+		msgType := dat["type"].(float64)
```

Comment:
this is true, changing to int

Reply:
tried changing to int, it actually has to be float64 because the data structure of the mapping of KV pairs maps keys to a float64 I think...it threw an error when trying to convert to int :/